apps/dashboard/app/models/script.rb

```diff
@@ -0,0 +1,78 @@
+# frozen_string_literal: true
+
+class Script
+  include ActiveModel::Model
+
+  attr_reader :title
+
+  attr_reader :id
+
+  class << self
+
+    def find(id, project_dir)
+      all(project_dir).select do |script|
+        # Rails.logger.debug("#{id} == #{script.id}")
+        id == script.id
+      end.first
+    end
+
+    def all(project_dir)
+      Dir.glob("#{Project.dataroot}/#{project_dir}/.ondemand/scripts/*.yml").map do |file|
+        Script.from_yaml(file, project_dir)
+      end
+    end
+
+    def from_yaml(file, project_dir)
+      contents = File.read(file)
+      opts = YAML.safe_load(contents).to_h
+      opts.merge!({ id: File.basename(file, '.yml') })
+      opts.merge!({ project_dir: project_dir.to_s })
+      new(opts)
```

Comment:
Again, it works, but it is too difficult to parse wrt maintainability.

What about:
```
def from_yaml(file, project_dir)
  contents = File.read(file)
  raw_opts = YAML.safe_load(contents)
  opts = raw_opts.to_h
  opts[:id] = File.basename(file, '.yml')
  opts[:project_dir] = project_dir.to_s
  new(opts)
end
```

Reply:
Sure

Reply:
I know it's minor, but I think those `merge!` statements are kinda hard to really grasp what this is doing. Maybe not for all, but I think it's just easier if you came back tot his code in 6 months to see those set explicitly is all.

Reply:
Yea some spacing could help us here. I've updated for the same.